crates/sui-framework/packages/sui-framework/sources/test/test_scenario.move

```diff
@@ -95,6 +109,109 @@ public struct TransactionEffects has drop {
     num_user_events: u64,
 }
 
+//
+// `TxContextBuilder` api
+//
+
+/// Create a new `TxContextBuilder` with the given `sender` address.
+/// Also provides default for all other fields.
+public fun ctx_builder_from_sender(sender: address): TxContextBuilder {
+    TxContextBuilder {
+        sender,
+        epoch: 0,
+        epoch_timestamp_ms: 0,
+        ids_created: 0,
+        rgp: option::some(700),
+        gas_price: 1_000,
+        gas_budget: 100_000_000,
+        sponsor: option::none(),
+    }
+}
+
+/// Create a `TxContextBuilder` from an existing `TxContext` in a `Scenario`.
+public fun ctx_builder(scenario: &Scenario): TxContextBuilder {
+    ctx_builder_from_context(&scenario.ctx)
+}
+
+/// Create a `TxContextBuilder` from an existing `TxContext`.
+public fun ctx_builder_from_context(ctx: &TxContext): TxContextBuilder {
+    TxContextBuilder {
+        sender: ctx.sender(),
+        epoch: ctx.epoch(),
+        epoch_timestamp_ms: ctx.epoch_timestamp_ms(),
+        ids_created: ctx.ids_created(),
+        rgp: option::some(ctx.reference_gas_price()),
+        gas_price: ctx.gas_price(),
+        gas_budget: ctx.gas_budget(),
+        sponsor: ctx.sponsor(),
+    }
+}
+
+/// Set the epoch for the `TxContextBuilder`.
+public fun set_epoch(mut builder: TxContextBuilder, epoch: u64): TxContextBuilder {
+    builder.epoch = epoch;
+    builder
+}
+
+/// Set the epoch timestamp in milliseconds for the `TxContextBuilder`.
+public fun set_epoch_timestamp(mut builder: TxContextBuilder, ms: u64): TxContextBuilder {
+    builder.epoch_timestamp_ms = ms;
+    builder
+}
+
+/// Set the ids created for the `TxContextBuilder`.
+public fun set_ids_created(mut builder: TxContextBuilder, ids_created: u64): TxContextBuilder {
+    builder.ids_created = ids_created;
+    builder
+}
+
+/// Set the reference gas price for the `TxContextBuilder`.
+public fun set_reference_gas_price(mut builder: TxContextBuilder, rgp: u64): TxContextBuilder {
+    builder.rgp = option::some(rgp);
+    builder
+}
+
+/// Set the reference gas price to `option::none()`.
+public fun unset_reference_gas_price(mut builder: TxContextBuilder): TxContextBuilder {
+    builder.rgp = option::none();
+    builder
+}
+
+/// Set the gas price for the `TxContextBuilder`.
+public fun set_gas_price(mut builder: TxContextBuilder, gas_price: u64): TxContextBuilder {
+    builder.gas_price = gas_price;
+    builder
+}
+
+/// Set the gas budget for the `TxContextBuilder`.
+public fun set_gas_budget(mut builder: TxContextBuilder, gas_budget: u64): TxContextBuilder {
+    builder.gas_budget = gas_budget;
+    builder
+}
+
+/// Set the sponsor for the `TxContextBuilder`.
+public fun set_sponsor(mut builder: TxContextBuilder, sponsor: Option<address>): TxContextBuilder {
```

Comment:
I feel like we should have done this change the same as RGP, feels a bit weird that they don't match